pkg/foundation/cerrors/fatal.go

```diff
@@ -0,0 +1,51 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cerrors
+
+import (
+	"fmt"
+)
+
+// FatalError is an error type that will differentiate these from other errors that could be retried.
+type FatalError struct {
+	Err error
+}
+
+// NewFatalError creates a new FatalError.
+func NewFatalError(err error) *FatalError {
+	return &FatalError{Err: err}
+}
+
+// Unwrap returns the wrapped error.
+func (f *FatalError) Unwrap() error {
+	if f == nil {
```

Comment:
I don't think we need this, `f` (which is the receiver for this method) can't be `nil` when the method is called

Reply:
@maha-hajja d'oh, you're right 🤦🏻‍♂️

Reply:
FYI, the receiver can actually be `nil` in Go if the method is called on a `nil` value 😉  https://go.dev/play/p/M1-eMFxvJgT